dask/distributed/worker.py

```diff
@@ -445,10 +461,20 @@ def collect(self, locations):
                 self.send_to_worker(worker, header, payload)
                 counter += 1
 
-            for i in range(counter):
-                queue.get()
+            msgs = [queue.get() for i in range(counter)]
+            for m in msgs:
+                if not m['got_key']:
```

Comment:
Having `msg['satus']` be either `'success'` or `'failed'` seemed to be asking for a boolean to me. I guess this leaves room for future statuses though.